### PR TITLE
[TT-6150] Add compressed/chunked response tests

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR extends the existing `compressed` response caching test to cover `chunked` response as well.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/test/add-chunked-body-cache
Commit: 82a51ba339bfa888638f3267361ac6dffe1f617a Add compressed/chunked response tests
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2869213009)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4239/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2869224755)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/test/add-chunked-body-cache
Commit: e0e60095b2dd8c89f0c36fae9b03dee998697cda Add compressed/chunked response tests
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2869326544)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4239/merge
Commit: e0e60095b2dd8c89f0c36fae9b03dee998697cda 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2869326681)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/test/add-chunked-body-cache
Commit: 978e6ee30380858a9e221151de1335fa171a7917 Add compressed/chunked response tests
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2869464176)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4239/merge
Commit: 978e6ee30380858a9e221151de1335fa171a7917 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2869464436)

File: gateway/mw_redis_cache_test.go [hunk 1]
```diff
@@ -8,62 +8,94 @@ import (
...
+		transferEncoding []string
+	}
 
-		ts.Run(t, []test.TestCase{
-			{Path: path, Code: 200, BodyMatch: "This is a compressed response"},
-			{Path: path, Code: 200, BodyMatch: "This is a compressed response"},
-		}...)
-	})
+	check := func(t *testing.T, p params) {
+		subCheck := func(t *testing.T, cachingActive bool, p params) {
```
Comment (titpetric) [inline gateway/mw_redis_cache_test.go:47]: invoke t.Helper() here

**Review by titpetric — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

File: gateway/mw_redis_cache_test.go (hunk 1 above)
Comment (furkansenharputlu) [reply gateway/mw_redis_cache_test.go:47]: As the assert functions and `t.Run` calls it, the trace is already printed.

**Comment by furkansenharputlu** [comment]
Manually merged to `release-4-lts`.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 40ee42f46cbfa7e4b81b707e6b589864e0281b32 Add compressed/chunked response tests (#4239)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2873325805)

**Pull request merged**